critiquebrainz/db/users.py

```diff
@@ -0,0 +1,44 @@
+from hashlib import md5
+import sqlalchemy
+from critiquebrainz import db
+
+
+def avatar(user):
+    url = "https://gravatar.com/avatar/{0}{1}"
+    if type(user) is dict:
+        if user['email'] and user['show_gravatar']:
+            return url.format(md5(user['email'].encode('utf-8')).hexdigest(),
+                              "?d=identicon&r=pg")
+        else:
+            return url.format(md5(str(user['id']).encode('utf-8')).hexdigest(),
+                              "?d=identicon")
+    return url.format(md5(user.encode('utf-8')).hexdigest(), "?d=identicon")
+
+
+def get_many_by_mb_username(usernames):
+    """Get information on users
+    Args:
+        usernames: A list of MusicBrainz usernames.
+    Returns:
+        All columns of 'user' table (list of dictionaries)
+        and avatar_url (Gravatar url).
+        If the 'usernames' variable is an empty list function returns it back.
+    """
+
+    with db.engine.connect() as connection:
+        if usernames:
+            result = connection.execute(sqlalchemy.text("""
+              SELECT id,
+                      display_name,
+                      email,
+                      created,
+                      musicbrainz_id,
+                      show_gravatar
+              FROM "user"
+              WHERE display_name IN :users"""), {'users': tuple(usernames)})
```

Comment:
And it would be better to make the username comparison case-insensetive, since they currently can be used like that. You can do something like `WHERE musicbrainz_id ILIKE ANY(...)`.